### ISSUE TITLE
A request to update dependency `Spglib.jl` to the latest version

### ISSUE THREAD

**Issue by singularitti** [thchr/Brillouin.jl#26]
Hi, thank you for using `Spglib.jl` in your code!
I've noticed that `Brillouin.jl` is currently using an [older version of `Spglib.jl`](https://github.com/thchr/Brillouin.jl/blob/1aa1e322bdb971837adff48547342fd8c32e17bf/Project.toml#L35). As the latest version of `Spglib.jl` includes several new features and bug fixes, it would be beneficial to update this dependency.

I've made several API changes in the latest updates. I understand these changes might require some adjustments in `Brillouin.jl` to ensure compatibility. Please let me know if you need any assistance with this update, I would be more than happy to help resolve the API changes and ensure `Brillouin.jl` works smoothly with the latest version of `Spglib.jl`.

**Comment by thchr** [comment]
It would be great to update it, definitely! Any help would be very welcome as I'm not very familiar with the API.

**Comment by singularitti** [comment]
Sure. I've just tried to use [your code](https://github.com/MineralsCloud/CrystallographyBase.jl/blob/8be374054710dbf906ad56eb825c88b3be1bce90/ext/BrillouinSpglibExt.jl), and it seems to be working (though I still need to check whether it gives the correct results).

**Comment by mfherbst** [comment]
I am updating DFTK to the latest Spglib now and I get an issue with the newest Spglib. Here is a typical stacktrace:
```
  MethodError: no method matching lu!(::CrystallographyCore.Lattice{Float64}, ::RowMaximum; check=false)
  Closest candidates are:
    lu!(::StridedMatrix{T}, ::RowMaximum; check) where T<:Union{Float32, Float64, ComplexF32, ComplexF64} at /opt/hostedtoolcache/julia/1.7.3/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:80
    lu!(::Union{Hermitian{T, S}, Symmetric{T, S}} where {T, S}, ::Union{NoPivot, RowMaximum}; check) at /opt/hostedtoolcache/julia/1.7.3/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:88
    lu!(::StridedMatrix{T} where T, ::Union{NoPivot, RowMaximum}; check) at /opt/hostedtoolcache/julia/1.7.3/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:134
    ...
  Stacktrace:
    [1] lu(A::CrystallographyCore.Lattice{Float64}, pivot::RowMaximum; check::Bool)
      @ LinearAlgebra /opt/hostedtoolcache/julia/1.7.3/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:279
    [2] det(A::CrystallographyCore.Lattice{Float64})
      @ LinearAlgebra /opt/hostedtoolcache/julia/1.7.3/x64/share/julia/stdlib/v1.7/LinearAlgebra/src/generic.jl:1559
    [3] irrfbz_path(cell::Spglib.SpglibCell{Float64, Float64, Int32, Float64})
      @ Brillouin.BrillouinSpglibExt ~/.julia/packages/Brillouin/Qx5t1/ext/BrillouinSpglibExt.jl:27
```

**Comment by mfherbst** [comment]
Would be great to get this resolved as this prevents me from updating Spglib in DFTK right now. Happy to help if people have some pointers.

**Comment by thchr** [comment]
@singularitti I would be grateful for your help or insights here: I'm not up-to-speed on the changes in Spglib.

**Comment by thchr** [comment]
The offending code for your error above seems to be here: 
https://github.com/thchr/Brillouin.jl/blob/1aa1e322bdb971837adff48547342fd8c32e17bf/ext/BrillouinSpglibExt.jl#L27C5-L30C8

I guess the types of `cell.lattice` and `dset.primitive_lattice` changed?

**Comment by singularitti** [comment]
Hi @mfherbst and @thchr, I'm sorry about the error. This is because I changed `Lattice` from a subtype of `AbstractMatrix` to a subtype of plain `AbstractLattice`. See https://github.com/MineralsCloud/CrystallographyCore.jl/pull/18

I did this because, to me, it does not make sense that a `Lattice` can be factorized or some other linear algebra API defined for matrices. I have never used them. I [asked a question on discourse](https://discourse.julialang.org/t/how-to-design-a-custom-array-like-type-for-physics-calculations-balancing-functionality-and-customization/103873), but not many people responded. Perhaps I was too vague about it. I only use `+`, `-`, `*`, `/` on `Lattice`s with a `Number`, as you can see [here in the code](https://github.com/MineralsCloud/CrystallographyCore.jl/blob/e9512932876161d130a1eec18e3fc7d466fcf1c5/src/lattice.jl#L154-L165).

For `det(lattice)`, I was using `cellvolume(lattice)` defined in [another package of mine](https://github.com/MineralsCloud/CrystallographyBase.jl/blob/7aeea2080cbacc26cd230a3a6d3a26dab89b24bb/src/volume.jl), where more methods are implemented.

Could you shed light on where `lu!` on a `Lattice` is useful? If you plan to use a `Matrix`, you could use `parent(lattice)` to get the underlying matrix.

That being said, it is strange this change has already affected you since this is only on the `main` branch, not yet been released. I was just trying to see how far I can go without having `Lattice` as a `Matrix`. Were you using the `main` branch?

I could split `CrystallographyCore` out as a dependency of `Spglib`, and put the old code directly into it. So that people would not be affected due to my constantly shifting ideas. Sorry, I was just trying to explore the power of Julia. But if people are using this code, I will be more careful. I should have told you, but this happened very recently.

**Comment by thchr** [comment]
No problem for my part: I think the issue only comes up on trying to update to a newer version of Spglib.jl, right? So should only affect developers and not current users of e.g., DFTK.jl.

More pragmatically: what should be done to update Brillouin.jl to a more recent/current version of Spglib.jl? Is it enough to just change the above code to `det(parent(cell.lattice))` and `det(parent(dset.primitive_lattice))` and updating the Spglib version?

**Comment by singularitti** [comment]
Hi @thchr, this change should only affect developers who are using the `main` branch of `Spglib`, the latest release of `Spglib` [v0.8.2](https://github.com/singularitti/Spglib.jl/releases/tag/v0.8.2) should work just fine. Besides, I recently [updated `spglib_jll`](https://github.com/JuliaPackaging/Yggdrasil/pull/7310) with the help of their team members, and they have officially noticed that we have a Julia API now (https://spglib.readthedocs.io/en/latest/interface.html).

> what should be done to update Brillouin.jl to a more recent/current version of Spglib.jl? Is it enough to just change the above code to det(parent(cell.lattice)) and det(parent(dset.primitive_lattice))

I think so, the old cold (`Lattice` as matrices) and the new code (`Lattice`s are just lattices) both define `parent`, which is a method that can be used out of abstract arrays. So whenever you use `parent(lattice)`, you get the underlying matrix, which stores lattice vectors as columns.

**Comment by mfherbst** [comment]
>  the latest release of Spglib v0.8.2 should work just fine.

Are you sure? I am using 0.8.2.

I added the `parent( )` function call in #27.

**Comment by singularitti** [comment]
> I am using 0.8.2.

Oh, I see. You were updating from 0.6/0.7 to 0.8, not 0.8 to `main`. As I mentioned, 0.8 is a breaking change since pre-0.8 releases are experimental. In v0.8 I probably transposed some matrices. I will help you go through this. Would you mind telling me which tests are causing errors in DFTK.jl?

**Comment by thchr** [comment]
Fixed by #27.
